include/swift/SILOptimizer/Utils/Differentiation/Common.h

```diff
@@ -33,6 +34,12 @@ namespace autodiff {
 /// This is being used to print short debug messages within the AD pass.
 raw_ostream &getADDebugStream();
 
+/// Given a function call site, gathers all of its actual results (both direct
+/// and indirect) in an order defined by its result type.
+void collectAllActualResultsInTypeOrder(
```

Comment:
I have a number of problems with this. I am pretty sure we already have something like this.